src/RazorSdk/SourceGenerators/RazorSourceGenerator.cs

```diff
@@ -223,19 +205,12 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
                 .Combine(importFiles.Collect())
                 .Combine(allTagHelpers)
                 .Combine(razorSourceGeneratorOptions)
-                .Combine(context.ParseOptionsProvider)
                 .Select(static (pair, _) =>
                 {
-                    var ((((sourceItem, imports), allTagHelpers), razorSourceGeneratorOptions), parserOptions) = pair;
+                    var (((sourceItem, imports), allTagHelpers), razorSourceGeneratorOptions) = pair;
```

Comment:
Yeah, its something we should think about. 

One API we considered was `TOut Combine<TOut, TLeft, TRight>(IncrementalValueProvider<TLeft>, IncrementalValuesProvider<TRight>, Func<TOut, TLeft, TRight>);`

Essentially allow you to select at the same time as combine, so you can build up a datamodel as you go. You can do it today via chaining, but it would be nice to make it easier.